lib/datadog/statsd/message_buffer.rb

```diff
@@ -17,9 +18,15 @@ def initialize(connection,
         @max_payload_size = max_payload_size
         @max_pool_size = max_pool_size
         @overflowing_stategy = overflowing_stategy
+        @flush_interval = flush_interval
+        @mon = Monitor.new
```

Comment:
It would be good to have a comment as to what this monitor protects

Reply:
I've written a comment in the commit https://github.com/DataDog/dogstatsd-ruby/pull/154/commits/a48591503ed5e57ba3fc882c1bba7b89c375b975.